sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalSensor.java

```
@@ -0,0 +1,98 @@
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.SensorDescriptor;
+import org.sonar.api.batch.sensor.issue.NewIssue;
+import org.sonar.api.batch.sensor.issue.NewIssueLocation;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.utils.command.Command;
+import org.sonar.api.utils.command.CommandExecutor;
+import org.sonar.api.utils.command.StreamConsumer;
+import org.sonar.api.utils.command.StringStreamConsumer;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+public class ExternalSensor implements Sensor {
+
+  private static final Logger LOG = Loggers.get(ExternalSensor.class);
+
+  private ExecutableBundle coreBundle;
+
+  public ExternalSensor(ExecutableBundle coreBundle) {
+    this.coreBundle = coreBundle;
+  }
+
+  @Override
+  public void describe(SensorDescriptor sensorDescriptor) {
+    sensorDescriptor.onlyOnLanguage(TypeScriptLanguage.KEY).name("External TypeScript Sensor").onlyOnFileType(InputFile.Type.MAIN);
+  }
+
+  @Override
+  public void execute(SensorContext sensorContext) {
+    String deployDestination = sensorContext.fileSystem().workDir().getPath();
+    coreBundle.deploy(deployDestination);
+    String projectSourcesRoot = sensorContext.fileSystem().baseDir().getPath();
+    Command command = coreBundle.createCommand(projectSourcesRoot, deployDestination);
+    try {
+      CommandExecutor commandExecutor = CommandExecutor.create();
+      StringStreamConsumer stdOut = new StringStreamConsumer();
+      StreamConsumer stdErr = new StringStreamConsumer(); // TODO create a different implementation that maps to analysisError
+      commandExecutor.execute(command, stdOut, stdErr, 1000);
+      Failure[] failures = new Gson().fromJson(stdOut.getOutput(), Failure[].class);
+      save(sensorContext, failures);
+    } catch (Exception e) {
+      LOG.error(String.format("Failed to run external process `%s`", command.toCommandLine()), e);
```

You should throw an exception to fail the analysis.